tests/cxx/explicit_instantiation2.cc

```diff
@@ -0,0 +1,99 @@
+//===------- explicit_instantiation2.cc - test input file for iwyu --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "explicit_instantiation-template_direct.h"
+#include "explicit_instantiation2-template_helpers.h"
+#include "explicit_instantiation2-template_short_direct.h"
+
+// These tests verify that a dependent explicit instantiation declaration is
+// always required when the main template is also needed, but not otherwise.
+//
+// Positive scenarios:
```

Comment:
Maybe these comments would have better locality next to the respective test case if possible?